File: humanre.client/interfaces/AddLeaveRequestObject.ts
export interface AddLeaveRequestObject {
    employeeId: number;
    startDate: string; 
    endDate: string; 
    numberOfDays: number;
    reason: string;
    isApproved?: boolean;
    isRejected?: boolean;
    isWithdrawn?: boolean;
    createdDate?: string; 
    modifiedDate?: string; 
    approvedById?: number | null;
    approvalDate?: string | null;
    rejectionReason?: string;
  }

File: HumanRe.Server/Models/AddLeaveRequestObject.cs
namespace HumanRe.Server.Models
{
    public class AddLeaveRequestObject
    {
        public int EmployeeId { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public int NumberOfDays { get; set; }
        public string Reason { get; set; }
        public bool IsApproved { get; set; } = false;
        public bool IsRejected { get; set; } = false;
        public bool IsWithdrawn { get; set; } = false;
        public DateTime CreatedDate { get; set; } = DateTime.UtcNow;
        public DateTime ModifiedDate { get; set; } = DateTime.UtcNow;

        public int? ApprovedById { get; set; } = null;
        public DateTime? ApprovalDate { get; set; } = null;
        public string RejectionReason { get; set; } = "";
    }
}
